### ISSUE TITLE
Filter out areas, devices and entities that aren't used

### ISSUE THREAD

**Issue by matthiasdebaat** [home-assistant/frontend#20242]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

For example on the automations page, it list all entities, that doesn't filter.
![CleanShot 2024-03-28 at 14 46 27@2x](https://github.com/home-assistant/frontend/assets/20203284/522e06d5-28bd-41c5-b27b-0c40eacf3352)

### Describe the behavior you expected

Filter out areas, devices and entities that aren't used

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

core-2024.4.0b0

**Comment by cloudbr34k84** [comment]
i was just thinking this! great idea

**Comment by bramkragten** [comment]
That is not possible atm, we don't know what is used.

**Comment by cloudbr34k84** [comment]
Yeah anything doesn't belong to an area or have labels or a category assigned?

**Comment by bramkragten** [comment]
The filter for entity and device are about what is used in an automation, we don't have that information currently. All we can currently do is ask for a specific entity/device in what automation it is used.
